llvm/include/llvm/ADT/StringRef.h

```diff
@@ -48,6 +52,8 @@ namespace llvm {
   /// lifetime extends past that of the StringRef. For this reason, it is not in
   /// general safe to store a StringRef.
   class LLVM_GSL_POINTER StringRef {
+    friend class llvm::sampleprof::ProfileFuncName;
```

Comment:
This is no-go. ADT should stay clear of its users, and ProfiledFuncName should not need to access private members.

Reply:
right. The ADT change is unwanted.